atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -124,13 +168,18 @@ def _render_mail(self, rebuild, success, canceled):
         ])
 
         endstate = None
-        if canceled:
-            endstate = 'canceled'
+        if auto_canceled or manual_canceled:
+            endstate = 'Canceled'
         else:
-            endstate = 'successful' if success else 'failed'
+            endstate = 'Succeeded' if success else 'Failed'
+
         url = None
-        if self.url and self.workflow.openshift_build_selflink:
-            url = urljoin(self.url, self.workflow.openshift_build_selflink + '/log')
+        try:
+            url = self._get_logs_url_from_build()
```

Comment:
Nitpick: This method actually is fetching the logs URL from a Koji task, not build. It might be a bit cleaner to simply encapsulate that and this try-except block in a single function: `_get_logs_url`.